collections.js

```diff
@@ -7,7 +7,7 @@ global.List = require("./list");
 global.Set = require("./set");
 global.Map = require("./map");
 global.MultiMap = require("./multi-map");
-global.WeakMap = require("weak-map");
+global.WeakMap = window.WeakMap || require("weak-map");
```

Comment:
Right, it won't avoid loads in dev but it will in deployment with mopped version and script loading


> On Feb 15, 2015, at 20:04, Kris Kowal <notifications@github.com> wrote:
> 
> In collections.js:
> 
> > @@ -7,7 +7,7 @@ global.List = require("./list");
> >  global.Set = require("./set");
> >  global.Map = require("./map");
> >  global.MultiMap = require("./multi-map");
> > -global.WeakMap = require("weak-map");
> > +global.WeakMap = window.WeakMap || require("weak-map");
> Ok. To be clear, this does avoid executing the module but does not affect loading or parsing in production, and executing does essentially what you have unrolled but only once when the module is executed first. I am not convinced that this renders a real performance gain.
> 
> —
> Reply to this email directly or view it on GitHub.
>